### DISCUSSION TITLE
GPU usage discussion

### DISCUSSION THREAD

**Discussion by Chotipon** [ultralytics/ultralytics#12642] [Q&A] [answered]
I'm working with yolov8 for detection task, from about 10 cameras source were input continuously. And the question is "Is it possible to reduce gpu usage to be about 10% ? " and "How to, if it possible" , i have tested on my 3060Ti which utilization is peak at 90-100% on yolov8n.pt model.

**Comment by glenn-jocher** [comment, marked as answer]
Reducing GPU usage to about 10% while handling inputs from 10 camera sources can be challenging due to the computational demands of detection tasks, especially with a powerful model like yolov8n.pt. Here are some suggestions you might consider:

1. **Reduce Model Size**: If you're not already, use a smaller model variant like YOLOv8n-lite which could reduce GPU load.
2. **Lower the Frame Rate**: Process fewer frames per second from each camera if real-time processing isn’t essential.
3. **Decrease Image Resolution**: Lower the `imgsz` during detection which reduces the amount of computation required.
4. **Batch Processing**: Combine frames from several cameras into batches to optimize GPU usage during inference.

Here’s a quick example of using a smaller model and reducing image size, which could help lower GPU utilization:

```bash
yolo detect predict model=yolov8n-lite.pt source='camera_input_path' imgsz=320
```

You might need to experiment with these adjustments to find a balance between performance and resource usage that suits your needs! 🚀
